signing/src/test/java/tech/pegasys/web3signer/signing/config/SignerLoaderTest.java

```diff
@@ -123,31 +127,35 @@ void wrongFileExtensionReturnsEmptySigner() throws IOException {
     final String filename = PUBLIC_KEY1 + ".nothing";
     createFileInConfigsDirectory(filename, PRIVATE_KEY1);
 
-    final List<ArtifactSigner> signerList =
-        Lists.newArrayList(new SignerLoader().load(configsDirectory, FILE_EXTENSION, signerParser));
+    final SecretValueResult<ArtifactSigner> result =
+        new SignerLoader().load(configsDirectory, FILE_EXTENSION, signerParser);
 
-    assertThat(signerList).isEmpty();
+    assertThat(result.getValues()).isEmpty();
+    assertThat(result.getErrorCount()).isZero();
     verifyNoMoreInteractions(signerParser);
   }
 
   @Test
   void failedParserReturnsEmptySigner() throws IOException {
     createFileInConfigsDirectory(PUBLIC_KEY1 + "." + FILE_EXTENSION, "NOT_A_VALID_KEY");
 
-    final List<ArtifactSigner> signerList =
-        Lists.newArrayList(new SignerLoader().load(configsDirectory, FILE_EXTENSION, signerParser));
+    final SecretValueResult<ArtifactSigner> result =
+        new SignerLoader().load(configsDirectory, FILE_EXTENSION, signerParser);
 
-    assertThat(signerList).isEmpty();
+    assertThat(result.getValues()).isEmpty();
   }
 
   @Test
   void failedWithDirectoryErrorReturnEmptySigner() throws IOException {
     final Path missingDir = configsDirectory.resolve("idontexist");
     createFileInConfigsDirectory(PUBLIC_KEY1 + "." + FILE_EXTENSION, PRIVATE_KEY1);
-    final List<ArtifactSigner> signerList =
-        Lists.newArrayList(new SignerLoader().load(missingDir, FILE_EXTENSION, signerParser));
+    final SecretValueResult<ArtifactSigner> result =
+        new SignerLoader().load(missingDir, FILE_EXTENSION, signerParser);
 
-    assertThat(signerList).isEmpty();
+    assertThat(result.getValues()).isEmpty();
+    // TODO: Should we return result with errorCount 1 if config directory is missing? Currently it
+    // is being treated as no error encountered
+    assertThat(result.getErrorCount()).isZero();
```

Comment:
I think your comment is correct, in the case where there is a directory error I would expect the an error to be registered

Reply:
Updated logic ... if config dir can't be read, the error count will not be zero.